sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/hash.scala

```diff
@@ -788,11 +799,21 @@ case class HiveHash(children: Seq[Expression]) extends HashExpression[Int] {
       $result = (int) ${HiveHashFunction.getClass.getName.stripSuffix("$")}.hashTimestamp($input);
      """
 
-  override protected def genHashString(input: String, result: String): String = {
-    val baseObject = s"$input.getBaseObject()"
-    val baseOffset = s"$input.getBaseOffset()"
-    val numBytes = s"$input.numBytes()"
-    s"$result = $hasherClassName.hashUnsafeBytes($baseObject, $baseOffset, $numBytes);"
+  override protected def genHashString(
```

Comment:
@cloud-fan I don't really understand the need for both `HiveHash` and default implementation in `HashExpression`. The diff seems to be that previous result is fed into hasher as seed for `HashExpression` but not for `Hive`, although I do not understand the reasoning for this.

I tried to mimic behaviour for collations but I will need detailed review from you on this.

Reply:
This is to follow the Hive behavior but we definitely do not want to follow hive for behavior of string with collation

Reply:
Ok, I see that `HiveHasher` doesn't even implement "seed" approach, so I am keeping it as is.